app/src/main/java/com/example/drawingactivity/DrawButtonView.kt

```diff
@@ -0,0 +1,69 @@
+package com.example.drawingactivity
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Color.*
+import android.graphics.Paint
+import android.graphics.Rect
+import android.util.AttributeSet
+import android.view.View
+import androidx.core.content.ContextCompat
+
+@SuppressLint("ClickableViewAccessibility")
+class DrawButtonView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : View(context, attrs, defStyle) {
+
+    var onDrawToolSet: ((Int) -> Unit)? = null
```

Comment:
kDoc

Reply:
Wasn't in use outside of the function so I deleted this for now. Original purpose was to track what tool was selected for future application, but I think I will solve that problem once I encounter it since my approach then might be completely different from what I had thought here.